checker.c

```diff
@@ -1,21 +1,164 @@
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+void BMS_setBMSStatus(bool bms_Status)
+{
+	BMS_parameters_attributes.bms_Status_b = bms_Status;			/*If any parameter out of range, set bms_Status_b as TRUE*/
+}
+
+void BMS_checkAttribute_Threshold_And_Trigger_Warning(float bms_attribute, float attribute_Warn_Lower_Threshold, float attribute_Warn_Upper_Threshold, char param[])
+{
+	if(bms_attribute < attribute_Warn_Lower_Threshold) 
+	{
+		//Trigger warning for Lower Threshold
+		BMS_printParameterStatus(param,"reaching Lower Threshold");
+	}
+	
+	if(bms_attribute > attribute_Warn_Upper_Threshold) 
+	{
+		//Trigger warning for Higher Threshold
+		BMS_printParameterStatus(param,"reaching Higher Threshold");
+	}
+}
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_batteryIsOk
+ */
+/*!    \brief   Function to test BMS parameters 
+ *
+ *
+ *     \param    	temperature -> temperature value
+					soc -> state of charge value
+					chargeRate -> charge rate value
+ *     \returns  int (0 or 1)
+ *
+ *//*------------------------------------------------------------------------*/
+int BMS_batteryIsOk(float temperature, float soc, float chargeRate) {
+	
+	BMS_parameters_attributes.temp_Status_b = BMS_checkAttribute_OutOfRange(temperature,BMS_ATTRIBUTE_TEMPERATURE_MIN_VALUE,BMS_ATTRIBUTE_TEMPERATURE_MAX_VALUE,"Temperature");
+	BMS_parameters_attributes.soc_Status_b = BMS_checkAttribute_OutOfRange(soc,BMS_ATTRIBUTE_SOC_MIN_VALUE,BMS_ATTRIBUTE_SOC_MAX_VALUE,"State of Charge");
+	BMS_parameters_attributes.chargeRate_Status_b = BMS_checkAttribute_OutOfRange(chargeRate,BMS_ATTRIBUTE_CHARGE_RATE_MIN_VALUE,BMS_ATTRIBUTE_CHARGE_RATE_MAX_VALUE,"Charge Rate");
+	
+	BMS_checkAttribute_Threshold_And_Trigger_Warning(temperature,BMS_ATTRIBUTE_TEMPERATURE_LOWER_THRESHOLD_WARN_VALUE,BMS_ATTRIBUTE_TEMPERATURE_UPPER_THRESHOLD_WARN_VALUE,"Temperature");
+	BMS_checkAttribute_Threshold_And_Trigger_Warning(soc,BMS_ATTRIBUTE_SOC_LOWER_THRESHOLD_WARN_VALUE,BMS_ATTRIBUTE_SOC_UPPER_THRESHOLD_WARN_VALUE,"State of Charge");
+	BMS_checkAttribute_Threshold_And_Trigger_Warning(chargeRate,BMS_ATTRIBUTE_CHARGE_RATE_LOWER_THRESHOLD_WARN_VALUE,BMS_ATTRIBUTE_CHARGE_RATE_UPPER_THRESHOLD_WARN_VALUE,"Charge Rate");
+	
+	if( TRUE == BMS_parameters_attributes.bms_Status_b) 
+	{
+		return 0;
+	}
+	else
+	{
+		return 1;
+	}
 }
 
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: main
+ */
+/*!    \brief   Function to test BMS parameters 
+ *
+ *
+ *     \param    none
+ *     \returns  int
+ *
+ *//*------------------------------------------------------------------------*/
 int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+  assert(BMS_batteryIsOk(25, 70, 0.7));			/*All values in range*/
+  assert(!BMS_batteryIsOk(50, 85, 0));			/*All values out of range*/
```

Comment:
Good addition of tests. The lower-breach condition is not reached by any of these tests?